### ISSUE TITLE
Replacement of `azurerm_cosmosdb_account` does not trigger replacement of dependent `azurerm_cosmosdb_gremlin_database`

### ISSUE THREAD

**Issue by ignaloidas** [hashicorp/terraform-provider-azurerm#20392]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.7

### AzureRM Provider Version

3.35.0

### Affected Resource(s)/Data Source(s)

azurerm_cosmosdb_account, azurerm_cosmosdb_gremlin_database

### Terraform Configuration Files

```hcl
data "azurerm_resource_group" "example" {
  name     = "main-group"
}

resource "azurerm_cosmosdb_account" "example" {
  name                = "test-account"
  location            = data.azurerm_resource_group.example.location
  resource_group_name = data.azurerm_resource_group.example.name
  offer_type          = "Standard"
  kind                = "GlobalDocumentDB"

  geo_location {
    location          = data.azurerm_resource_group.example.location
    failover_priority = 0
  }

  consistency_policy {
    consistency_level = "Session"
  }

  capabilities {
    name = "EnableGremlin"
  }

  tags = local.tags
}

resource "azurerm_cosmosdb_gremlin_database" "example" {
  name                = "test"
  resource_group_name = data.azurerm_resource_group.example.name
  account_name        = azurerm_cosmosdb_account.example.name
}
```

### Debug Output/Panic Output

```shell
No relevant debug output - all commands succeed until you try modifying the `azurerm_cosmosdb_gremlin_database` resource.
```

### Expected Behaviour

On recreation of Cosmos DB account, the gremlin database would be re-created as well.

### Actual Behaviour

Cosmos DB account gets re-created, but the database does not, even though it still exists in terraform state.

### Steps to Reproduce

1. `terraform apply`
2. Update the configuration to force replacement of `azurerm_cosmosdb_account`, for example adding 
```
capabilities {
    name = "EnableServerless"
  }
```
3. `terraform apply`
4. Now `azurerm_cosmosdb_gremlin_database` exists in terraform state, even though there is no such azure resource backing it.

**Comment by sinbai** [comment]
@ignaloidas thanks for opening this issue. For the expected behavior "_On recreation of Cosmos DB account, the gremlin database would be re-created as well._", I would like to explain that is this case the resource `azurerm_cosmosdb_gremlin_database` is recreated  depending on whether the `account_name` is changed. I assume that the `account_name` in your case is not changed, right? If yes , I think  it's by design that `azurerm_cosmosdb_gremlin_database` is not recreated since the `account_name` is not changed.

**Comment by ignaloidas** [comment]
Indeed, the `account_name` has not changed. Seems like this isn't possible with the way terraform currently works. That's annoying, but I guess this probably should be documented? It made a fair bit of mess when it happened to me, and I had to mess with state to fix it, which definitely isn't nice.

**Comment by sinbai** [comment]
@ignaloidas thanks for your reply. For " _I guess this probably should be documented?_ ", does this mean that documenting changing `account_name` will force the creation of a new resource to be documented in Terraform doc? If yes, "Changing this forces a new resource to be created" is documented [here](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/cosmosdb_gremlin_database#account_name). "changing this" means the `account_name` is changed not the resource `azurerm_cosmosdb_account` is changed.

**Comment by ignaloidas** [comment]
I think there should be a note on `azurerm_cosmosdb_gremlin_database` (and probably similar resources) that if a `azurerm_cosmosdb_account` they are using is re-created without a name change, it can end up in a broken state.

**Comment by sinbai** [comment]
Thanks for your feedback. BTW, in this case, have you tried [refreshing ](https://developer.hashicorp.com/terraform/cli/commands/refresh)the state file to ensure sync?

**Comment by ignaloidas** [comment]
I've forgot that it exists, but I'm not sure if it would've helped in my particular case. In my case the recreation of `azurerm_cosmosdb_account` was combined with updates of `azurerm_cosmosdb_gremlin_database` which led to the update step breaking in a weird way, and new, broken `azurerm_cosmosdb_gremlin_database` resources being actually created.

**Comment by fuzzykiller** [comment]
Having just encountered this with an `azurerm_cosmosdb_sql_database`, I’d say this applies to all Cosmos DB child resources. I would appreciate a general fix here.
